onnxruntime/core/graph/graph.cc

```diff
@@ -3128,4 +3389,139 @@ std::ostream& operator<<(std::ostream& out, const Graph& graph) {
+  // TODO: We could decide that an ORT model is load only even in a full build,
+  // and in InferenceSession::Initialize skip partitioning and running optimizers.
+  ORT_RETURN_IF_ERROR(graph->Resolve());
+#else
+  // probably nothing required here. validate with model that has nested subgraphs.
+#endif
+
+  return Status::OK();
+}
+
+Status Graph::LoadFromOrtFormat(const onnxruntime::experimental::fbs::Graph& fbs_graph,
+                                Graph& parent_graph, const Node& parent_node,
+                                const logging::Logger& logger, std::unique_ptr<Graph>& graph) {
+  // can't use make_unique as we're calling a private ctor
+  graph.reset(new Graph(parent_graph.owning_model_,
+                        parent_graph.domain_to_version_, &parent_graph, &parent_node,
+                        logger));
+
+  return graph->LoadFromOrtFormat(fbs_graph);
+}
+
+Graph::Graph(const Model& owning_model,
+             const std::unordered_map<std::string, int>& domain_to_version,
+             Graph* parent_graph, const Node* parent_node,
+             const logging::Logger& logger)
+    : owning_model_(owning_model),
+      graph_proto_(&deserialized_proto_data_),
+#if !defined(ORT_MINIMAL_BUILD)
+      schema_registry_(std::make_shared<SchemaRegistryManager>()),
+#endif
+      domain_to_version_(domain_to_version),
+      parent_graph_(parent_graph),
+      parent_node_(parent_node),
+      logger_(logger),
+      is_loaded_from_model_file_(true) {  // true as the Graph isn't manually constructed from scratch
+}
+
+common::Status Graph::LoadFromOrtFormat(const onnxruntime::experimental::fbs::Graph& fbs_graph) {
+  // load initializers
+  auto fbs_initializers = fbs_graph.initializers();
+  if (fbs_initializers) {
+    name_to_initial_tensor_.reserve(fbs_initializers->size());
+    for (const auto* fbs_tensor : *fbs_initializers) {
+      ORT_RETURN_IF_NOT(nullptr != fbs_tensor, "fbs_tensor cannot be null");
+      TensorProto* initializer = deserialized_proto_data_.add_initializer();
+      ORT_RETURN_IF_ERROR(experimental::utils::LoadInitializerOrtFormat(*fbs_tensor, *initializer));
+      name_to_initial_tensor_[initializer->name()] = initializer;
+    }
+  }
+
+  // NodeArgs
+  auto fbs_node_args = fbs_graph.node_args();
+  if (fbs_node_args) {
+    node_args_.reserve(fbs_node_args->size());
+    for (const auto* fbs_value_info : *fbs_node_args) {
+      ORT_RETURN_IF_NOT(nullptr != fbs_value_info, "fbs_value_info cannot be null");
+      NodeArgInfo node_arg_info;
+      ORT_RETURN_IF_ERROR(experimental::utils::LoadValueInfoOrtFormat(*fbs_value_info, node_arg_info));
+      std::unique_ptr<NodeArg> n(new NodeArg(std::move(node_arg_info)));
+      node_args_[n->Name()] = std::move(n);
+    }
+  }
+
+  // Nodes
+  nodes_.resize(fbs_graph.max_node_index());
+  auto* fbs_nodes = fbs_graph.nodes();
+  if (fbs_nodes != nullptr) {
+    for (const auto* fbs_node : *fbs_nodes) {
```

Comment:
Should maybe add a comment saying this is theoretically possible. e.g. subgraph for If node may just have a Constant node that gets lifted to an initializer.